### ISSUE TITLE
Firefox notifications in wrong position and off-screen

### ISSUE THREAD

**Issue by johninman** [emersion/mako#93]
I'm running sway 1.0-beta.1-160-g4a219818 and mako 1.1-6 from the AUR, and experiencing a strange issue. Notifications from most applications or via `notify-send` work fine and appear in the bottom right corner like I've specified, but web notifications from Firefox (email and the like) display in the top left corner with the left half off the screen. I'm not sure if the bug lies with mako itself or is a symptom of some other issue but I figured this was as good a place to start as any.

**Comment by emersion** [comment]
Can you share a screenshot?

Do the Firefox notifications look like the other mako notifications?

**Comment by vilhalmer** [comment]
>I wouldn't be surprised if it checks for a notification daemon on run and sets its own solution if none are found.

Yep, this is the case. Chrome does the same, it's a limitation of the D-Bus notification protocol.

**Comment by emersion** [comment]
>mako doesn't support buttons at all yet, I suspect that Firefox is generating its own for some reason

Another possibility is that maybe Firefox checks capabilities and if the notification has buttons (and the server doesn't support these) it decides to create its own.

But yeah in general starting mako while other apps are running doesn't work very well.

**Comment by xPMo** [comment]
I was trying to solve this recently and found [an ArchLinux thread](https://bbs.archlinux.org/viewtopic.php?id=252375)

It links [this file in Gecko's source](https://github.com/mozilla/gecko-dev/blob/581d9b3815d363cd2a6ef53c1f92bf536a23b5a8/toolkit/system/gnome/nsAlertsIconListener.cpp#L321)

If the notification it wants do display doesn't have actions, it will use libnotify if it a server responds to the notify_get_server_caps call. If the notification does have actions, it looks for the "actions" capability.
